src/darwin/Framework/CHIP/EndpointDescription/MTRAccessGrant.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRBaseClusters.h>
+#import <Matter/MTRDefines.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * An access grant, which can be represented as an entry in the Matter Access
+ * Control cluster.
+ */
+MTR_NEWLY_AVAILABLE
+@interface MTRAccessGrant : NSObject <NSCopying>
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/**
+ * Grant access at the provided level to a specific node on the fabric.  The
+ * provided nodeID must be an operational node identifier.
+ */
+- (nullable instancetype)initWithNodeID:(NSNumber *)nodeID privilege:(MTRAccessControlEntryPrivilege)privilege error:(NSError * __autoreleasing *)error;
+
+/**
+ * Grant access to any node on the fabric that has a matching CASE Authenticated
+ * Tag in its operational certificate.  The provided caseAuthenticatedTag must
+ * be a 32-bit unsigned integer with lower 16 bits not 0, per the Matter
+ * specification.
+ */
+- (nullable instancetype)initWithCASEAuthenticatedTag:(NSNumber *)caseAuthenticatedTag privilege:(MTRAccessControlEntryPrivilege)privilege error:(NSError * __autoreleasing *)error;
+
+/**
+ * Grant access to any node on the fabric that is communicating with us via
+ * group messages sent to the given group.  The provided groupID must be a valid
+ * group identifier in the range 1-65535.
+ */
+- (nullable instancetype)initWithGroupID:(NSNumber *)groupID privilege:(MTRAccessControlEntryPrivilege)privilege error:(NSError * __autoreleasing *)error;
+
+/**
+ * Grant access to any node on the fabric, as long as it's communicating with us
+ * over a unicast authenticated channel.
+ */
+- (instancetype)initForAllNodesWithPrivilege:(MTRAccessControlEntryPrivilege)privilege;
+
+/**
+ * The matter access control subject ID that access has been granted for.  Nil
+ * when access has been granted for all subjects (e.g. via initForAllNodesWithPrivilege).
+ */
+@property (nonatomic, copy, readonly, nullable) NSNumber * subjectID;
```

Comment:
It might be nicer to have separate `nodeID` and `groupID` properties (one of which will be nil), and then we wouldn't need the `authenticationMode` property?

Reply:
Two things:

1. For internal use, the subjectID is in fact what I want.
2. In future, if we add the ability to specify an ACL as "allow access level X for any group message", we want to be able to express that.  Right now I am not adding that since we have no use case for it, but Matter ACLs do support that, and we might want it at some point.  So we need the authenticationMode no matter what, to tell apart the "any CASE connection" and "any groupcast" cases.

Reply:
Hm, shoving the nodeID and the groupID into the same property just because they happen to be both numbers just seems like a bit of a code smell.